contracts/optimism/TokenRateOracle.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2022 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.10;
+
+import {ITokenRateOracle} from "../token/interfaces/ITokenRateOracle.sol";
+
+/// @author kovalgek
+/// @notice Oracle for storing token rate.
+contract TokenRateOracle is ITokenRateOracle {
+
+    error NotAnOwner(address caller);
+    error IncorrectRateTimestamp();
+
+    /// @notice wstETH/stETH token rate.
+    uint256 private tokenRate;
+
+    /// @notice L1 time when token rate was pushed. 
+    uint256 private rateL1Timestamp;
+
+    /// @notice A bridge which can update oracle.
+    address public immutable bridge;
+
+    /// @notice An updater which can update oracle.
+    address public immutable tokenRateUpdater;
+
+    /// @notice A time period when token rate can be considered outdated.
+    uint256 public immutable heartbeatPeriodTime;
+
+    /// @param bridge_ the bridge address that has a right to updates oracle.
+    /// @param tokenRateUpdater_ address of oracle updater that has a right to updates oracle.
+    /// @param heartbeatPeriodTime_ time period when token rate can be considered outdated.
+    constructor(address bridge_, address tokenRateUpdater_, uint256 heartbeatPeriodTime_) {
+        bridge = bridge_;
+        tokenRateUpdater = tokenRateUpdater_;
+        heartbeatPeriodTime = heartbeatPeriodTime_;
+    }
+
+    /// @inheritdoc ITokenRateOracle
+    function latestRoundData() external view returns (
+        uint80 roundId_,
+        int256 answer_,
```

Comment:
should we named it `tokenRate`? answer is always a token rate...
the same for `latestAnswer` -> `latestTokenRate`

Reply:
this due to keeping compatibility with the chainlink interface